libkernelflinger/security_efi.c

```diff
@@ -98,51 +96,92 @@ EFI_STATUS set_device_security_info(__attribute__((unused)) IN void *security_da
 	BOOTLOADER_SEED_PROTOCOL *bls;
 	UINT8 key_count = BLS_MAX_RPMB_KEY;
 	UINT8 rpmb_keys[BLS_MAX_RPMB_KEY][RPMB_KEY_SIZE];
-#ifdef USE_TPM
 	UINT8 rpmb_key[RPMB_KEY_SIZE];
 	UINTN key_size = sizeof(rpmb_key);
-#endif
 	UINT8 i;
 
 	// Set the fixed RPMB key
 	if (is_live_boot()) {
+		// For USB live boot case, always use one fixed RPMB key and seed, and do not store the lock/unlock state.
+		set_secure_storage_type(SECURE_STORAGE_NONE);
+		return set_rpmb_derived_key(fixed_rpmb_keys, RPMB_KEY_SIZE, 1);
+	}
+
+	if (!is_platform_secure_boot_enabled()) {
+		// For secure boot disabled case, always use one fixed RPMB key and seed, and store the lock/unlock state to EFI variable.
+		set_secure_storage_type(SECURE_STORAGE_EFI_VAR);
 		// For USB live boot case, always use one fixed RPMB key.
 		return set_rpmb_derived_key(fixed_rpmb_keys, RPMB_KEY_SIZE, 1);
 	}
 
+	// Try BLS
```

Comment:
This function is over complicated for no reasons.
The C following original construction is cumbersome, confusing and create a increased indentation for just no reason.

```
EFI_STATUS set_device_security_info()
{
[...]
	bls = get_bls_proto();
        if (bls) {
                /* do a lot of stuff */
                return;
        }

        ret = function_call();
        return ret;
} 
```

```
EFI_STATUS set_device_security_info()
{
[...]
	bls = get_bls_proto();
        if (!bls)
                return function_call();

        /* do a lot of stuff */
} 
```

I know it was not your code but your piling up with the same unacceptable pattern because you have to fit in in this bad logic. Why don't you try to rework the code and then make your changes.

As a general question: Do you seriously have code reviews because this code should not have entered as-is !